### PR TITLE
Dry-run JSON debug output 1

### PR DESCRIPTION
**Pull request by isseis**
This pull request implements and documents the dry-run debug JSON output feature, completing Phases 1 and 2 of the project. It introduces new data structures, helper functions, and collection logic for environment variable inheritance and final environment analysis, along with comprehensive tests and integration checks. The most important changes are summarized below:

### Data Model Additions and Updates

* Added new constants (`ResourceTypeGroup`, `OperationAnalyze`) and updated comments in `internal/runner/resource/types.go` to support resource and operation typing for debug output.
* Defined new data structures: `DebugInfo`, `InheritanceAnalysis`, `FinalEnvironment`, and `EnvironmentVariable` in `internal/runner/resource/types.go`, with appropriate JSON tags and documentation. [[1]](diffhunk://#diff-1530741f122f63d88804d01ec0cd8ddcf3967988f0551c829814ff115ef00699L63-R65) [[2]](diffhunk://#diff-1530741f122f63d88804d01ec0cd8ddcf3967988f0551c829814ff115ef00699L78-R83) [[3]](diffhunk://#diff-1530741f122f63d88804d01ec0cd8ddcf3967988f0551c829814ff115ef00699L96-R99)
* Added a `DebugInfo` pointer field to the `ResourceAnalysis` struct for attaching debug information.

### Data Collection Logic

* Implemented the new `internal/runner/debug/collector.go` file, including functions to collect inheritance analysis and final environment data, with support for different detail levels and sensitive value masking. [[1]](diffhunk://#diff-1530741f122f63d88804d01ec0cd8ddcf3967988f0551c829814ff115ef00699L195-R197) [[2]](diffhunk://#diff-fd71a4722c89c038f3f681b889061be98ac1ec17d414a73bed5b4efe828481c8R1-R171)
* Added helper functions for safe slice handling, set operations, and variable extraction to support robust data collection.

### Serialization and Testing

* Implemented custom JSON marshal/unmarshal methods for `InheritanceMode` in `internal/runner/runnertypes/config.go`, with error handling for invalid values.
* Added and updated unit tests for serialization and data collection logic in `internal/runner/runnertypes/config_test.go` and `internal/runner/debug/collector_test.go`, covering all detail levels and edge cases. [[1]](diffhunk://#diff-1530741f122f63d88804d01ec0cd8ddcf3967988f0551c829814ff115ef00699L138-R148) [[2]](diffhunk://#diff-1530741f122f63d88804d01ec0cd8ddcf3967988f0551c829814ff115ef00699L246-R255) [[3]](diffhunk://#diff-1530741f122f63d88804d01ec0cd8ddcf3967988f0551c829814ff115ef00699L286-R292)

### Integration and Milestones

* Verified integration by running `make test` and `make lint` successfully for each phase, ensuring all new structures and functions work correctly and existing tests pass (no regressions). [[1]](diffhunk://#diff-1530741f122f63d88804d01ec0cd8ddcf3967988f0551c829814ff115ef00699L159-R162) [[2]](diffhunk://#diff-1530741f122f63d88804d01ec0cd8ddcf3967988f0551c829814ff115ef00699L303-R306) [[3]](diffhunk://#diff-1530741f122f63d88804d01ec0cd8ddcf3967988f0551c829814ff115ef00699L661-R662)
* Marked project milestones and commits for Phase 1 and Phase 2 as complete in the implementation plan documentation. [[1]](diffhunk://#diff-1530741f122f63d88804d01ec0cd8ddcf3967988f0551c829814ff115ef00699L180-R182) [[2]](diffhunk://#diff-1530741f122f63d88804d01ec0cd8ddcf3967988f0551c829814ff115ef00699L324-R326)

These changes lay the groundwork for robust dry-run debugging and JSON output, with well-structured data models, reliable collection logic, and strong test coverage.

### CODE REVIEW

File: internal/runner/debug/collector.go [hunk 1]
```diff
@@ -0,0 +1,171 @@
+//go:build test
```
[the remark this answers is not in the document]
Comment (isseis) [reply internal/runner/debug/collector.go]: done

File: internal/common/slice_test.go [hunk 2]
```diff
@@ -5,6 +5,200 @@ import (
...
+
+		for i, expected := range input {
+			if result[i] != expected {
+				t.Errorf("expected result[%d] = %q, got %q", i, expected, result[i])
+			}
+		}
+	})
+}
+
+func TestSetDifference(t *testing.T) {
```
[the remark this answers is not in the document]
Comment (isseis) [reply internal/common/slice_test.go]: done

File: internal/common/slice_test.go [hunk 3]
```diff
@@ -5,6 +5,200 @@ import (
 	"testing"
 )
 
+func TestSafeStringSlice(t *testing.T) {
```
[the remark this answers is not in the document]
Comment (isseis) [reply internal/common/slice_test.go]: done
